From 02a756b98b5204f2ae57ee88eac16b37d8c9ef8e Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:08 -0400
Subject: [scsi] qla2xxx: Reset nic_core_reset_owner on moving from COLD to READY for ISP8044

Message-id: <1408391817-6389-21-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87914
O-Subject: [PATCH RHEL 7.1 e-stor 20/69] qla2xxx: Reset nic_core_reset_owner on moving from COLD to READY for ISP8044
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sawan Chandak <sawan.chandak@qlogic.com>

scsi commit id 3746078fb022ddba836bbb1283e0e6e6b21af2c7

Signed-off-by: Sawan Chandak <sawan.chandak@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_nx2.c b/drivers/scsi/qla2xxx/qla_nx2.c
index 6f509f8..cc3522f 100644
--- a/drivers/scsi/qla2xxx/qla_nx2.c
+++ b/drivers/scsi/qla2xxx/qla_nx2.c
@@ -1860,7 +1860,7 @@ qla8044_device_state_handler(struct scsi_qla_host *vha)
    goto exit;
   case QLA8XXX_DEV_COLD:
    rval = qla8044_device_bootstrap(vha);
-   goto exit;
+   break;
   case QLA8XXX_DEV_INITIALIZING:
    qla8044_idc_unlock(ha);
    msleep(1000);
-- 
1.7.1